File: src/Coffee/CoffeeBL/Ordered_Coffee.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoffeeBL
{
    public enum coffee_type
    {
        None=0,
        mill = 1,
        brewed = 2
    }
    public class Ordered_Coffee
    {
        public Ordered_Coffee()
        {
            this.quantity = 1;
            this.coffee = new Coffee();
            this.syroup = new Syroup();
        }
        public Coffee coffee { get; set; }
        public Syroup syroup { get; set; }
        public coffee_type type { get; set; }

        public int quantity { get; set; }
        public float actual_price {
            get
            {
                return quantity * coffee.price + syroup.price;
            }
        
        }

        

    }
}

File: src/CafeleWeb/src/app/OrderedCoffee.ts
import { Coffee } from './coffee';
import { Syroup } from './syroup';

export class OrderedCoffee {
    coffee:Coffee;
    syroup:Syroup;
    nrCoffees: number;
    /**
     *
     */
    constructor() {
       this.syroup = null;
       this.coffee = null;   
           
    }

    public TotalPrice(): number 
    {
        let priceCoffee=0;
        if(this.coffee != null){
            priceCoffee=this.coffee.price;
        }
        if(this.syroup != null)
        {
            priceCoffee += this.syroup.price;
        }
        return this.nrCoffees * priceCoffee;
    }
}
